ansible/roles/redpanda_broker/templates/configs/tiered_storage.j2

```diff
@@ -1,11 +1,20 @@
 cluster:
-  cloud_storage_access_key: THISVALUENOTUSED
-  cloud_storage_bucket: {{ tiered_storage_bucket_name if tiered_storage_bucket_name is defined }}
-  cloud_storage_enable_remote_read: true
-  cloud_storage_enable_remote_write: true
-  cloud_storage_region: {{ cloud_storage_region if cloud_storage_region is defined }}
-  cloud_storage_secret_key: THISVALUENOTUSED
-  cloud_storage_credentials_source: aws_instance_metadata
-  # cloud_storage_enabled must be after other cloud_storage parameters
-  cloud_storage_enabled: {{ true if tiered_storage_bucket_name is defined and tiered_storage_bucket_name|d('')|length > 0 else false }}
-
+{% if cloud_storage_credentials_source is defined %}
+    cloud_storage_credentials_source: {{ cloud_storage_credentials_source if cloud_storage_credentials_source is defined }}
+{% else %}
+    cloud_storage_access_key: {{ cloud_storage_access_key if cloud_storage_access_key is defined }}
+    cloud_storage_secret_key: {{ cloud_storage_secret_key if cloud_storage_secret_key is defined }}
+{% endif %}
+{% if cloud_storage_enable_remote_read is defined %}
+    cloud_storage_enable_remote_read: {{ cloud_storage_enable_remote_read }}
+{% endif %}
+{% if cloud_storage_enable_remote_write is defined %}
+    cloud_storage_enable_remote_write: {{ cloud_storage_enable_remote_write }}
+{% endif %}
+    cloud_storage_region: {{ cloud_storage_region if cloud_storage_region is defined }}
```

Comment:
region question: what are your thoughts in providing some guardrails around validating the existence of the supplied region? May help with preventing the user from doing surprising things (typos, etc)

Reply:
I think there'd be a lot of complexity that would be hard to resolve without some sort of plugin. 

Like, for aws correct region naming is us-west-2
For gcp correct region naming is us-west2

So we'd have to check the provider, then check that they've correctly hyphenated from there. 

And then there's azure: westus2

I can see that being advantageous in the long run, but in the short run it'd just eat cycles to prevent something that will fail pretty quick and get resolved anyway.

Reply:
Got it! Definitely not required for this, I was thinking/typing out loud.